backend/app/api/routers/cipher.py

```
@@ -95,8 +95,107 @@ def delete_memory(
     if memory.get("user_id") != user_id:
         raise HTTPException(status_code=403, detail="Access denied")
 
-    # Memory deletion not yet implemented — return honest 501 instead of lying
-    raise HTTPException(status_code=501, detail="Memory deletion not yet implemented")
+    try:
+        db.delete_memory(memory_id)
+    except AttributeError:
+        raise HTTPException(status_code=501, detail="Memory deletion not supported by current database backend")
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)